internal/provider/aws/provider_test.go

```diff
@@ -0,0 +1,4906 @@
+					mockec2Client.EXPECT().
+						DescribeAddresses(ctx, gomock.AssignableToTypeOf(&ec2.DescribeAddressesInput{})).
+						DoAndReturn(func(_ context.Context, input *ec2.DescribeAddressesInput, _ ...func(*ec2.Options)) (*ec2.DescribeAddressesOutput, error) {
+							switch len(input.Filters) {
+							case 3:
+								Expect(input.Filters).To(ConsistOf(
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyManaged)),
+										Values: []string{"true"},
+									},
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyClusterID)),
+										Values: []string{clusterID},
+									},
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyExternalIPName)),
+										Values: []string{externalIP.Name},
+									},
+								))
+								return &ec2.DescribeAddressesOutput{}, nil
+							case 4:
+								Expect(input.Filters).To(ConsistOf(
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyManaged)),
+										Values: []string{"true"},
+									},
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyClusterID)),
+										Values: []string{clusterID},
+									},
+									types.Filter{
+										Name:   aws.String(fmt.Sprintf("tag:%s", TagKeyExternalIPName)),
+										Values: []string{externalIP.Name},
+									},
+									types.Filter{
+										Name:   aws.String("allocation-id"),
+										Values: []string{allocationID},
+									},
+								))
+								return &ec2.DescribeAddressesOutput{
+									Addresses: []types.Address{
+										{
+											AllocationId:  aws.String(allocationID),
+											AssociationId: aws.String(associationID),
+										},
+									},
+								}, nil
+							default:
+								return nil, fmt.Errorf("unexpected number of filter: %d", len(input.Filters))
+							}
+						}).Times(2)
+					mockec2Client.EXPECT().
+						AllocateAddress(ctx, gomock.AssignableToTypeOf(&ec2.AllocateAddressInput{})).
+						DoAndReturn(func(_ context.Context, input *ec2.AllocateAddressInput, _ ...func(*ec2.Options)) (*ec2.AllocateAddressOutput, error) {
+							Expect(input.Domain).To(Equal(types.DomainTypeVpc))
+							Expect(input.TagSpecifications).To(HaveLen(1))
+							Expect(input.TagSpecifications[0].ResourceType).To(Equal(types.ResourceTypeElasticIp))
+							Expect(input.TagSpecifications[0].Tags).To(ConsistOf(
+								types.Tag{
+									Key:   aws.String(string(TagKeyManaged)),
+									Value: aws.String("true"),
+								},
+								types.Tag{
+									Key:   aws.String(string(TagKeyClusterID)),
+									Value: aws.String(clusterID),
+								},
+								types.Tag{
+									Key:   aws.String(string(TagKeyExternalIPName)),
+									Value: aws.String(externalIP.Name),
+								},
+								types.Tag{
+									Key:   aws.String(string(TagKeyInstanceID)),
+									Value: aws.String(instanceID),
+								},
```

Comment:
I think this would be better written with a single matcher.
```suggestion
							Expect(input.TagSpecifications).To(ConsistOf(
								MatchFields(IgnoreExtras, Fields{
									"ResourceType": Equal(types.ResourceTypeElasticIp),
									"Tags": ConsistOf(
										types.Tag{
											Key:   aws.String(string(TagKeyManaged)),
											Value: aws.String("true"),
										},
										types.Tag{
											Key:   aws.String(string(TagKeyClusterID)),
											Value: aws.String(clusterID),
										},
										types.Tag{
											Key:   aws.String(string(TagKeyExternalIPName)),
											Value: aws.String(externalIP.Name),
										},
										types.Tag{
											Key:   aws.String(string(TagKeyInstanceID)),
											Value: aws.String(instanceID),
										},
									),
								}),
							))
```